pkg/ipcache/metadata.go

```diff
@@ -224,7 +224,7 @@ func (ipc *IPCache) InjectLabels(ctx context.Context, modifiedPrefixes []netip.P
 			// kvstore and via the k8s control plane. If the new
 			// security identity is the same as the one currently
 			// being used, then no need to update it.
-			if oldID.ID == newID.ID {
+			if oldID.ID == newID.ID && prefixInfo.Source() == oldID.Source {
```

Comment:
Ah interesting. Previously this code assumed for one that if the identity is the same, then the source of that identity is the same. However as the comment demonstrates immediately above, that may not be true. Secondly I think it also implicitly added an assumption that if the eventual result of the ipcache update here is going to map a specific IP to a specific Identity, then if the identity is the same, we don't need to propagate the event to the other subsystems because the ultimate result in the datapath is the same. For the `UpdatePolicyMaps()` call below I think that's true - technically there is nothing to do if the source is different. However for the main ipcache structure update below as part of the `entriesToReplace` logic, that layer knows about `source` and uses it for some protections against ipcache entry deletion etc. Therefore, that layer really should be informed about a change in source. :+1:

Reply:
While I think we could potentially avoid the extra policymap notification for this case, this probably doesn't have much of an impact and likely just results in a little extra processing to figure out that it's a no-op. That's acceptable IMO.